hail/src/main/scala/is/hail/expr/ir/Infer.scala

```diff
@@ -61,15 +61,7 @@ object Infer {
         val tbs = coerce[TStruct](old.typ)
         TStruct(fields.map { id => (id, tbs.field(id).typ) }: _*)
       case InsertFields(old, fields) =>
-        fields.foldLeft(old.typ) { case (t, (name, a)) =>
```

Comment:
oh god

Reply:
this is way worse than just a map. It's reallocating all of the cached stuff on TStruct 10000 times! missing bits, the field names array, the types array, oh dear.